### PR TITLE
Expose Daml-LF parser to trigger simulations

### PR DESCRIPTION
**Pull request by carlpulley-da**
To simplify writing trigger simulations we:
- expose the Daml-LF (testing) parser to trigger simulations
- add in a convenience method to parse Daml-LF strings into SValues (via Values)
- add in a convenience method to simplify setup of simulations

### CODE REVIEW

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala [hunk 1]
```diff
@@ -107,6 +113,47 @@ abstract class TriggerMultiProcessSimulation extends AsyncWordSpec with Abstract
...
+      val (client, actAs) = Await.result(setup, simulationConfig.simulationSetupTimeout)
+      val ledger = controllerContext.spawn(LedgerProcess.create(client), "ledger")
+
+      controllerContext.watch(ledger)
+
+      spawnTriggers(client, ledger, actAs, controllerContext)
+    }
+  }
+
+  protected def safeSValueFromLf(lfValue: String): Either[String, SValue] = {
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala:139]: Method to simplify defining trigger initial user state values etc.

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala [hunk 2]
```diff
@@ -107,6 +113,47 @@ abstract class TriggerMultiProcessSimulation extends AsyncWordSpec with Abstract
       actAs,
     )
   }
+
+  protected def withLedger(
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala:117]: Method to simplify setting up simulation environments

File: triggers/tests/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        "@maven//:org_scalaz_scalaz_core",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":test-utils",
+        "//canton/it-lib",
+        "//daml-lf/data",
+        "//daml-lf/interpreter",
+        "//daml-lf/language",
+        "//daml-lf/parser",
```
Comment (remyhaemmerle-da) [inline triggers/tests/BUILD.bazel:255]: where this is used ?

File: triggers/tests/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+    "//bazel_tools:scala.bzl",
+    "da_scala_binary",
+    "da_scala_library",
+    "da_scala_test_suite",
+    "lf_scalacopts_stricter",
+)
+load("@build_environment//:configuration.bzl", "sdk_version")
+
+DAML_LF_VERSIONS = [
+    "1.dev",
```
Comment (remyhaemmerle-da) [inline triggers/tests/BUILD.bazel:14]: please only test one version of LF

File: triggers/tests/BUILD.bazel (hunk 3 above)
Comment (carlpulley-da) [reply triggers/tests/BUILD.bazel:255]: This is used to define the function `unsafeSValueFromLf` in the `TriggerMultiProcessSimulation` abstract class.

Comment (remyhaemmerle-da) [reply triggers/tests/BUILD.bazel:255]: I see fair enough. 
Maybe a good idea to write such an example.

Comment (carlpulley-da) [reply triggers/tests/BUILD.bazel:255]: Absolutely. FYI, this example is pulled from the draft PR https://github.com/digital-asset/daml/blob/63c09b82dc28a21b4176a7336482c8e452b4aaf7/triggers/simulations/scala/Contention.scala.

I've intentionally split out this change to try and simplify the review process. Apologies if this has caused issues.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks.

**Pull request merged**